### PR TITLE
support index parameters in call holder

### PR DESCRIPTION
**Pull request by reiase**
Signed-off-by: jie.hou <jie.hou@zilliz.com>

### CODE REVIEW

**Comment by reiase** [comment]
@Chiiizzzy

File: towhee/hparam/hyperparameter.py [hunk 1]
```diff
@@ -162,43 +162,48 @@ def __call__(self, default: Any = None) -> Any:
...
 
     @staticmethod
-    def default_callback(path, *arg, **kws):
-        return (path, arg, kws)
+    def default_callback(path, index, *arg, **kws):
+        return (path, index, arg, kws)
 
-    def __init__(self, callback: Callable=None, path=None):
+    def __init__(self, callback: Callable = None, path=None, index=None):
         super().__init__()
```
Comment (filip-halt) [inline towhee/hparam/hyperparameter.py:183]: super still needed?

Comment (reiase) [reply towhee/hparam/hyperparameter.py:183]: no, I will drop this line.

**Comment by Chiiizzzy** [comment]
\lgtm

**Pull request merged**